tests/integ/test_session.py

```diff
@@ -679,19 +679,20 @@ def test_sql_simplifier_disabled_on_session(db_parameters):
     reason="reading server side parameter is not supported in local testing",
 )
 def test_cte_optimization_enabled_on_session(db_parameters):
+    default_value = True
```

Comment:
This test seems going to be flaky if we have to turn the flag off again. we just want to test whether the default value is the value "_PYTHON_SNOWPARK_USE_CTE_OPTIMIZATION_STRING", right? why not just testing switching the value of _PYTHON_SNOWPARK_USE_CTE_OPTIMIZATION_STRING and see if the new_session.cte_optimization_enabled is the correct value or not

Reply:
good point, fixed